### PR TITLE
CMake: Improve GStreamer Inclusion

### PR DESCRIPTION
**Pull request by HTRamsey**
Significantly better method of handling GStreamer in CMake

### CODE REVIEW

**Comment by doquangquan93** [comment]
That's great. I tried it out. It works perfectly

File: src/VideoManager/CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +7,16 @@ qt_add_library(VideoManager STATIC
     VideoManager.h
 )
 
+# option(QGC_ENABLE_VIDEOSTREAMING "Enable video streaming" ON)
```
Comment (DonLakeFlyer) [inline src/VideoManager/CMakeLists.txt:10]: Why leave in commented out?

File: src/VideoManager/VideoReceiver/GStreamer/GStreamer.cc [hunk 2]
```diff
@@ -182,18 +182,25 @@ GStreamer::initialize(int argc, char* argv[], int debuglevel)
...
-    qgcputenv("GST_REGISTRY_REUSE_PLUGIN_SCANNER", "no", "");
-    qgcputenv("GST_PLUGIN_SCANNER", "/usr/lib/x86_64-linux-gnu", "/gstreamer1.0/gstreamer-1.0/gst-plugin-scanner");
-    qgcputenv("GST_PTP_HELPER_1_0", "/usr/lib/x86_64-linux-gnu", "/gstreamer1.0/gstreamer-1.0/gst-ptp-helper");
-    qgcputenv("GTK_PATH", "/usr", "");
-    qgcputenv("GIO_EXTRA_MODULES", "/usr/lib/x86_64-linux-gnu", "/gio/modules");
-    qgcputenv("GST_PLUGIN_SYSTEM_PATH_1_0", "/usr/lib/x86_64-linux-gnu", "/gstreamer-1.0");
-    qgcputenv("GST_PLUGIN_SYSTEM_PATH", "/usr/lib/x86_64-linux-gnu", "/gstreamer-1.0");
-    qgcputenv("GST_PLUGIN_PATH_1_0", currentDir, "../lib");
-    qgcputenv("GST_PLUGIN_PATH", currentDir, "../lib");
+    // const QString currentDir = QCoreApplication::applicationDirPath();
```
Comment (DonLakeFlyer) [inline src/VideoManager/VideoReceiver/GStreamer/GStreamer.cc:194]: Why leave in commented out?

File: src/VideoManager/VideoReceiver/GStreamer/CMakeLists.txt [hunk 3]
```diff
@@ -37,20 +36,19 @@ if(IOS)
...
+        gstqml6gl
         Settings
         VideoReceiver
 )
 
-target_include_directories(GStreamerReceiver PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
-
 target_compile_definitions(GStreamerReceiver PUBLIC QGC_GST_STREAMING)
-if(CMAKE_BUILD_TYPE STREQUAL "Release")
+if(MACOS AND CMAKE_BUILD_TYPE STREQUAL "Release")
```
Comment (DonLakeFlyer) [inline src/VideoManager/VideoReceiver/GStreamer/CMakeLists.txt:52]: How come Mac only?

File: src/VideoManager/VideoReceiver/GStreamer/GStreamer.cc (hunk 2 above)
Comment (HTRamsey) [reply src/VideoManager/VideoReceiver/GStreamer/GStreamer.cc:194]: I was still messing with those when I had time. I'm having a hard time getting it perfect for all systems

**Pull request merged**